### PR TITLE
Support for compound devices and "grandchildren"

### PR DESCRIPTION
**Pull request by dljsjr**
This refactor furthers the generalizations done previously for devices that provide multiple services to allow for not only mapping of services to capabilities or components, but also multiple device records.

This allows for us to support devices such as the Dymera up/down light.

Other changes:

- I found myself frequently introducing bugs related to using the Hue Bridge ID (MAC Address) as a Device ID, and the Bridge Device ID as a Hue Bridge ID. We renamed all instances of `bridge_id` referencing the MAC to `bridge_network_id` to try and alleviate the confusion.

- Button support is further generalized to support two-button models, which covers the In-Wall Switch Module in the 2-gang configuration.

### CODE REVIEW

File: drivers/SmartThings/philips-hue/src/handlers/lifecycle_handlers.lua [hunk 1]
```diff
@@ -53,11 +53,21 @@ function LifecycleHandlers.device_added(driver, device, ...)
...
+        driver, device, device.parent_device_id or device:get_field(Fields.PARENT_DEVICE_ID)
+      )
 
       local key = parent_bridge and parent_bridge:get_field(HueApi.APPLICATION_KEY_HEADER)
       local bridge_ip = parent_bridge and parent_bridge:get_field(Fields.IPV4)
       local bridge_id = parent_bridge and parent_bridge:get_field(Fields.BRIDGE_ID)
+      log.trace(true,
+        st_utils.stringify_table(
+          {parent_bridge and parent_bridge.label, key, bridge_ip, bridge_id},
+          "device added bridge deets",
```
Comment (tpmanley) [inline drivers/SmartThings/philips-hue/src/handlers/lifecycle_handlers.lua:67]: 😎

**Review by tpmanley — APPROVED** [review]

File: drivers/SmartThings/philips-hue/src/disco/button.lua [hunk 2]
```diff
@@ -63,7 +63,7 @@ local function _do_update(api_instance, device_service_info, bridge_id, resource
   end
 
   if type(cache) == "table" then
-    cache[resource_id] = button_remote_description
+    cache[button_remote_description.id] = button_remote_description
```
Comment (cjswedes) [inline drivers/SmartThings/philips-hue/src/disco/button.lua:66]: will `button_remove_description.id` always be nil? it seems like it, but maybe this is being set on line 49?

**Review by cjswedes — APPROVED** [review]
There's a lot going on here.

**Comment by dljsjr** [comment]
> There's a lot going on here.

Yeah. Adjust the assumption of a Hue `device` service only exposing a single `light` service was a deep surgery 😢

**Review by NoahCornell — APPROVED** [review]

**Pull request merged**